sonic-slave-buster/Dockerfile.j2

```diff
@@ -40,11 +48,88 @@ RUN echo "deb [arch=arm64] http://deb.debian.org/debian buster main contrib non-
         echo 'deb [arch=arm64] http://ftp.debian.org/debian buster-backports main' >> /etc/apt/sources.list && \
         echo "deb [arch=arm64] http://packages.trafficmanager.net/debian/debian buster main contrib non-free" >> /etc/apt/sources.list && \
         echo "deb [arch=arm64] http://packages.trafficmanager.net/debian/debian buster-updates main contrib non-free" >> /etc/apt/sources.list
+{%- elif CONFIGURED_ARCH == "armhf" and CROSS_BUILD_ENVIRON == "y" %}
+RUN dpkg --add-architecture armhf
+RUN apt-get update 
+RUN apt-get install -y crossbuild-essential-armhf
+RUN apt-get install -y gcc-arm-linux-gnueabihf
+ENV $(dpkg-architecture -a$(CONFIGURED_ARCH))
+
+RUN apt-mark hold crossbuild-essential-armhf
+RUN apt-mark hold g++-arm-linux-gnueabihf
+RUN apt-mark hold g++-8-arm-linux-gnueabihf
+RUN apt-mark hold gcc-arm-linux-gnueabihf
+RUN apt-mark hold gcc-8-arm-linux-gnueabihf
+
+ARG arch=armhf
+ARG CROSS_CC=arm-linux-gnueabihf-gcc
+ARG CROSS_CXX=arm-linux-gnueabihf-g++
+ARG PYTHON_CROSS_PLATFORM=linux_armv7l
+{%- elif CONFIGURED_ARCH == "arm64" and CROSS_BUILD_ENVIRON == "y" %}
+RUN dpkg --add-architecture arm64
+RUN apt-get update
+RUN apt-get install -y crossbuild-essential-arm64
+RUN apt-get install -y gcc-aarch64-linux-gnu
+ENV $(dpkg-architecture -a$(CONFIGURED_ARCH))
+
+RUN apt-mark hold crossbuild-essential-arm64
+RUN apt-mark hold g++-aarch64-linux-gnu
+RUN apt-mark hold g++-8-aarch64-linux-gnu
+RUN apt-mark hold gcc-aarch64-linux-gnu
+RUN apt-mark hold gcc-8-aarch64-linux-gnu
+ARG arch=arm64
+ARG CROSS_CC=aarch64-linux-gnu-gcc
+ARG CROSS_CXX=aarch64-linux-gnu-g++
+ARG PYTHON_CROSS_PLATFORM=linux_aarch64
 {%- endif %}
```

Comment:
Could you unified the branches? In theory, it should be applicable to any cross compilation, not limited to armhf/arm64.

Reply:
Don't understand what you mean by unifying the branches ?
Can you give an example please.

Reply:
The armhf branch is quite similar as arm64 branch. In theory, it should be similar to any arch. Could you unify for any general arch? For example, let's assume we have an `arm32` arch, can we easily adapt?

Reply:
`arch` and `gcc_arch` can come from `DEB_HOST_ARCH` and `DEB_HOST_GNU_TYPE` respectively. For example:

```
$ dpkg-architecture -a armhf -q DEB_HOST_ARCH
armhf
$ dpkg-architecture -a armhf -q DEB_HOST_GNU_TYPE
dpkg-architecture: warning: specified GNU system type arm-linux-gnueabihf does not match CC system type x86_64-linux-gnu, try setting a correct CC environment variable
arm-linux-gnueabihf
$ dpkg-architecture -a arm64 -q DEB_HOST_ARCH
arm64
$ dpkg-architecture -a arm64 -q DEB_HOST_GNU_TYPE
dpkg-architecture: warning: specified GNU system type aarch64-linux-gnu does not match CC system type x86_64-linux-gnu, try setting a correct CC environment variable
aarch64-linux-gnu
```

I don't know about `PYTHON_CROSS_PLATFORM` though.